### ISSUE TITLE
modelimport, TrainedModels.VGG16.decodePredictions bug?

### ISSUE THREAD

**Issue by tomthetrainer** [deeplearning4j/deeplearning4j#2789]
I think there is a bug in decodePredictions

I load the model, pass it a lion and here is my output code

INDArray[] output = vgg16.output(false,image);
// get decodePredictions output
System.out.println(TrainedModels.VGG16.decodePredictions(output[0]));

// get output array and sort it with indices
INDArray[] sorted = Nd4j.sortWithIndices(output[0],1,false);
System.out.println(sorted[0].data());

When classifying a lion 

You typed/Users/tomhanlon/tensorflow/vgg16/keras-model-zoo/deep-learning-models/images/lion.jpeg
Predictions for batch  :
	0.000000%, whiptail
	0.000000%, whiptail
	0.000000%, whiptail
	0.000000%, whiptail
	0.000000%, whiptail
## Yet the output is correct it is just being decoded incorrectly

First five from sorted output
291.0,231.0,274.0,230.0,260.0

Looking the labels up manually
https://gist.github.com/yrevar/942d3a0ac09ec9e5eb3a
 291: 'lion, king of beasts, Panthera leo',
231: 'collie',
274: 'dhole, Cuon alpinus',

Okay so lion was not so good, but still not a whiptail. 
Here is elephant

Predictions for batch  :
	0.000000%, European_gallinule
	0.000000%, European_gallinule
	0.000000%, European_gallinule
	0.000000%, European_gallinule
	0.000000%, European_gallinule
[386.0,101.0,385.0,51.0,354.0,

Manual lookup..
386: 'African elephant, Loxodonta africana',
101: 'tusker',
385: 'Indian elephant, Elephas maximus',
51: 'triceratops',
354: 'Arabian camel, dromedary, Camelus dromedarius',

**Comment by tomthetrainer** [comment]
Here is a gist of my code

https://gist.github.com/tomthetrainer/d86cc2c2b9dbc9da0756c980d0cae59f

**Comment by tomthetrainer** [comment]
Strangely enough, it gets this dog right, every time, but it misses the others. 
File Path: /Users/tomhanlon/tensorflow/vgg16/keras-model-zoo/deep-learning-models/images/dog.jpeg
You typed/Users/tomhanlon/tensorflow/vgg16/keras-model-zoo/deep-learning-models/images/dog.jpeg
Predictions for batch  :
	34.196171%, Labrador_retriever
	24.212713%, golden_retriever
	12.633617%, redbone
	11.096197%, Rhodesian_ridgeback
	4.622131%, bloodhound
[208.0,207.0,168.0,159.0,163.0

208 is lab, 207 is golden.. etc

**Comment by turambar** [comment]
Hmmmm, did you try running inference with the imported model _before_ serializing it?

**Comment by tomthetrainer** [comment]
@turambar Yeah I did a long time ago, I worked around it. It is a bug.  

This works

```
// Get output
            INDArray[] output = vgg16.output(false, image);
            // Sort output for top 5
            INDArray[] sorted = Nd4j.sortWithIndices(output[0], 1, false);
            // sorted map for results
            Map<Float, String> map = new TreeMap<Float, String>(Collections.reverseOrder());
            // Get top 5
            for (int i = 0; i < 5; i++) {
                // Get prediction percent
                Float prediction = sorted[1].getFloat(i) * 100;
                // extract label for prediction
                String Label = ImageNetLabels.getLabel(sorted[0].getInt(i));
                // put both in map
                map.put(prediction,Label);
            }
            
            System.out.println(map);
```

**Comment by eraly** [comment]
I'll look at it. It shouldn't be too hard to fix.

**Comment by eraly** [comment]
I'm closing this out as I traced this to a bug unrelated to model import. index reduce is broken.
